yroots/IntervalChecks.py

```diff
@@ -77,18 +86,47 @@ def __init__(self,a,b):
         self.interval_results["Too Deep"] = []
         self.total_area = np.prod(self.b-self.a)
         self.current_area = 0.
-        self.polishing = False
         self.tick = 0
+        
+        #For polishing code
+        self.polishing = False
+        self.polish_intervals = []
+        self.polish_num = 0
+        self.polish_interval_num = -1
+        self.polish_a = np.array([])
+        self.polish_b = np.array([])
 
-    def check_interval(self, coeff, approx_tol, a, b):
+    def add_polish_intervals(self, polish_intervals):
+        ''' Add the intervals that polishing will be run on.
+
+        Parameters
+        ----------
+        polish_intervals : list
+            The intervals polishing will be run on.
+        '''
+        self.polishing = True
+        self.polish_intervals = polish_intervals
+        self.polish_num += 1
+        self.polish_interval_num = -1
+        
+    def start_polish_interval(self):
+        '''Get the tracking ready to track the next polished interval
+        '''
+        #self.tick = 99 #So it will print right away.
+        self.polish_interval_num += 1
+        self.polish_a, self.polish_b = self.polish_intervals[self.polish_interval_num]
+        self.total_area = np.prod(self.polish_b-self.polish_a)
+        self.current_area = 0.
```

Comment:
This tracks the intervals that polishing will be run on. Basically to keep the print statement that says how for the code has come up to date.